ktor-server/ktor-server-plugins/ktor-server-auth/jvm/src/io/ktor/server/auth/SessionAuth.kt

```diff
@@ -106,43 +126,29 @@ public inline fun <reified T : Principal> Authentication.Configuration.session(
 
 /**
  * Provides ability to authenticate users via sessions. It is important to have
- * specified [SessionAuthenticationProvider.Configuration.validate] and
- * [SessionAuthenticationProvider.Configuration.challenge] in the lambda
+ * specified [SessionAuthenticationProvider.Config.validate] and
+ * [SessionAuthenticationProvider.Config.challenge] in the lambda
  * to get it work property
  */
-public inline fun <reified T : Any> Authentication.Configuration.session(
+public inline fun <reified T : Any> AuthenticationConfig.session(
     name: String? = null,
-    configure: SessionAuthenticationProvider.Configuration<T>.() -> Unit
+    configure: SessionAuthenticationProvider.Config<T>.() -> Unit
 ) {
-    val provider = SessionAuthenticationProvider.Configuration(name, T::class).apply(configure).buildProvider()
-
-    provider.pipeline.intercept(AuthenticationPipeline.CheckAuthentication) { context ->
-        val session = call.sessions.get<T>()
-        val principal = session?.let { provider.validator(call, it) }
-
-        if (principal != null) {
-            context.principal(principal)
-        } else {
-            val cause =
-                if (session == null) AuthenticationFailedCause.NoCredentials
-                else AuthenticationFailedCause.InvalidCredentials
-
-            context.challenge(SessionAuthChallengeKey, cause) {
-                provider.challenge(this, principal)
-                if (!it.completed && call.response.status() != null) {
-                    it.complete()
-                }
-            }
-        }
-    }
-
+    val provider = SessionAuthenticationProvider.Config(name, T::class).apply(configure).buildProvider()
     register(provider)
 }
 
+/**
+ * A context for [SessionAuthChallengeFunction]
+ */
+public class SessionChallengeContext(
```

Comment:
Could you tell me why do we need this class?

Reply:
To keep source compatibility and have `call: ApplicationCall` as a receiver